[CI] TransformIndexerTests#testRetentionPolicyExecution fails

**Build scan**:

7x latest: https://gradle-enterprise.elastic.co/s/ilmtqogzqdieg
master latest: https://gradle-enterprise.elastic.co/s/dqqxcbuu6sfhw

**Repro line**:

./gradlew ':x-pack:plugin:transform:test' --tests "org.elasticsearch.xpack.transform.transforms.TransformIndexerTests.testRetentionPolicyExecution" \
  -Dtests.seed=2D5237B136E4CF5A \
  -Dtests.security.manager=true \
  -Dtests.locale=ar-LY \
  -Dtests.timezone=Israel \
  -Druntime.java=8

**Reproduces locally?**:

No

**Applicable branches**:

master, 7.x

**Failure history**:

Several instances since Feb 8th.

**Failure excerpt**:

```
java.lang.AssertionError: expected:<STARTED> but was:<INDEXING>
	at __randomizedtesting.SeedInfo.seed([2D5237B136E4CF5A:F1DE71FBD55EC5FD]:0)
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:118)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at org.elasticsearch.xpack.transform.transforms.TransformIndexerTests.testRetentionPolicyExecution(TransformIndexerTests.java:332)
```

--- a/x-pack/plugin/transform/src/test/java/org/elasticsearch/xpack/transform/transforms/TransformIndexerTests.java
+++ b/x-pack/plugin/transform/src/test/java/org/elasticsearch/xpack/transform/transforms/TransformIndexerTests.java
@@ -329,7 +329,7 @@ public class TransformIndexerTests extends ESTestCase {
 
             indexer.start();
             assertTrue(indexer.maybeTriggerAsyncJob(System.currentTimeMillis()));
-            assertEquals(indexer.getState(), IndexerState.INDEXING);
+            assertThat(indexer.getState(), oneOf(IndexerState.INDEXING, IndexerState.STARTED));
 
             assertBusy(() -> assertEquals(1L, indexer.getLastCheckpoint().getCheckpoint()), 5, TimeUnit.SECONDS);
 

